common/djangoapps/student/tests/tests.py

```diff
@@ -487,6 +485,41 @@ def test_linked_in_add_to_profile_btn_with_certificate(self):
         )
         self.assertContains(response, expected_url)
 
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    @ddt.data((ModuleStoreEnum.Type.mongo, 1), (ModuleStoreEnum.Type.split, 3))
+    @ddt.unpack
+    def test_dashboard_metadata_caching(self, modulestore_type, expected_mongo_calls):
+        """
+        Check that the student dashboard makes use of course metadata caching.
+
+        The first time the student dashboard displays a specific course, it will
+        make a call to the module store. After that first request, though, the
+        course's metadata should be cached as a CourseOverview.
+
+        Arguments:
+            modulestore_type (ModuleStoreEnum.Type): Type of modulestore to create
+                test course in.
+            expected_mongo_calls (int >=0): Number of MongoDB queries expected for
+                a single call to the module store.
```

Comment:
Note to self: Add comment here explaining why `check_mongo_calls(0)` is important